student.rb

```diff
@@ -2,9 +2,12 @@
 
 # Student class extends person class
 class Student < Person
+  attr_reader :classroom
+
   def initialize(age, name, _parent, parent_permission, classroom)
     super(age, name, parent_permission)
-    @class_room = classroom
+    @classroom = classroom
+    classroom.student.push(self) unless classroom.students.include?(self)
```

Comment:
**[OPTIONAL]** Rather than initializing the `classroom` here, you can define a separate method let's say `classroom` which will accept a `classroom` argument and have both lines there.